java/ql/src/experimental/Security/CWE/CWE-094/SpringViewManipulation.ql

```diff
@@ -13,7 +13,15 @@ import java
 import SpringViewManipulationLib
 import DataFlow::PathGraph
 
-from DataFlow::PathNode source, DataFlow::PathNode sink, SpringViewManipulationConfig conf
-where conf.hasFlowPath(source, sink)
+from
+  DataFlow::PathNode source, DataFlow::PathNode sink, SpringViewManipulationConfig conf,
+  StringFlowConfig s
+where
+  thymeleafIsUsed() and
+  conf.hasFlowPath(source, sink) and
+  not s.hasFlow(_, sink.getNode()) and
+  not exists(AddExpr e | e.getLeftOperand().(StringLiteral).getRepresentedString() = "redirect:" |
```

Comment:
need to account for other forms of __concatenation__ such as format strings, StringBuffer and StringWriter to mention a few

Reply:
after reading StringFlowConfig, this may be already accounted for. It would be nice to add testcases though

Reply:
@pwntester the other forms of concatenations would be handled by the data flow config. it is just the direct addition of the string which QL fails to recognize.